### PR TITLE
Param validation actors

### LINKED ISSUE #114: Parameter Validation on /actors API
Apply parameter validations as defined in :
https://github.com/retaildevcrews/helium/blob/master/docs/ParameterValidation.md

References: https://github.com/retaildevcrews/helium/issues/253

### PR DESCRIPTION
**Pull request by jyotsnaravikumar**
## Type of PR

- [ ] Documentation changes
- [ X] Code changes
- [ ] Test changes
- [ ] CI-CD changes
- [ ] GitHub Template changes

## Purpose of PR
Parameter Validation implementation on /actor 

## Issues Closed or Referenced

- Closes #114

### CODE REVIEW

File: src/main/java/com/microsoft/cse/helium/app/controllers/ActorsController.java [hunk 1]
```diff
@@ -3,116 +3,138 @@
...
+    }
 
+    Integer pageNo = 0;
     if (pageNumber.isPresent() && !StringUtils.isEmpty(pageNumber.get())) {
-      if (pageNumber.get() >= 1 && pageNumber.get() <= Constants.MAX_PAGE_COUNT) {
-        tmpPageNumber = pageNumber.get();
+      if (!validator.isValidPageNumber(pageNumber.get())) {
+        logger.error("Invalid pageNumber parameter");
+        MultiValueMap<String, String> headers = new HttpHeaders();
+        headers.put("content-type", Arrays.asList("text/plain"));
```
Comment (sanjeevdwivedi) [inline src/main/java/com/microsoft/cse/helium/app/controllers/ActorsController.java]: Shouldn't this be Content-type (note upper case)?

File: src/main/java/com/microsoft/cse/helium/app/controllers/ActorsController.java [hunk 2]
```diff
@@ -3,116 +3,138 @@
...
-      if (tmpPageSize < 1) {
-        tmpPageSize = Constants.DEFAULT_PAGE_SIZE;
-      } else if (tmpPageSize > Constants.MAX_PAGE_SIZE) {
-        tmpPageSize = Constants.MAX_PAGE_SIZE;
+    Integer pageSz = Constants.DEFAULT_PAGE_SIZE;
+    if (pageSize.isPresent() && !StringUtils.isEmpty(pageSize.get())) {
+      if (!validator.isValidPageSize(pageSize.get())) {
+        logger.error("Invalid pageSize parameter");
+        MultiValueMap<String, String> headers = new HttpHeaders();
+        headers.put("content-type", Arrays.asList("text/plain"));
```
Comment (sanjeevdwivedi) [inline src/main/java/com/microsoft/cse/helium/app/controllers/ActorsController.java]: Capitailization on content-type

File: src/main/java/com/microsoft/cse/helium/app/utils/ParameterValidator.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+  /** isValidActorId. */
+  public Boolean isValidActorId(String actorId) {
+    Pattern p = Pattern.compile(validActorRegex);
+    return p.matcher(actorId).matches();
+  }
+
+  /** isValidSearchQuery. */
+  public Boolean isValidSearchQuery(String query) {
+    Boolean isValid = true;
```
Comment (sanjeevdwivedi) [inline src/main/java/com/microsoft/cse/helium/app/utils/ParameterValidator.java]: The above code can be simplified a bit. Whenever we do parameter validation, we should aim to return immediately on hitting an error. Essentially, stop on fail, so the following is pretty standard way of writing such code.
isValidSearchQuery(String query) {
    if(...) {
        return false 
   }
  return true
}

File: src/main/java/com/microsoft/cse/helium/app/controllers/ActorsController.java (hunk 1 above)
Comment (jofultz) [reply src/main/java/com/microsoft/cse/helium/app/controllers/ActorsController.java]: Isn't there a MediaType.TEXT_PLAIN constant to use?  If so, please change it to use that.

File: src/main/java/com/microsoft/cse/helium/app/controllers/ActorsController.java (hunk 2 above)
Comment (jofultz) [reply src/main/java/com/microsoft/cse/helium/app/controllers/ActorsController.java]: Same here with regard to MediaType.

Comment (jyotsnaravikumar) [reply src/main/java/com/microsoft/cse/helium/app/controllers/ActorsController.java]: Fixed

File: src/main/java/com/microsoft/cse/helium/app/utils/ParameterValidator.java (hunk 3 above)
Comment (jyotsnaravikumar) [reply src/main/java/com/microsoft/cse/helium/app/utils/ParameterValidator.java]: This is done to maintain consistency with other validation methods where true or false is calculated from multiple decision trees.

**Pull request merged**